crates/sui-indexer/src/store/pg_indexer_store.rs

```diff
@@ -271,6 +272,103 @@ impl PgIndexerStore {
         })
     }
 
+    fn persist_objects_snapshot_chunk(
+        &self,
+        objects: Vec<ObjectChangeToCommit>,
+    ) -> Result<(), IndexerError> {
+        let guard = self
+            .metrics
+            .checkpoint_db_commit_latency_objects_snapshot_chunks
+            .start_timer();
+        let mut mutated_objects: Vec<StoredObjectSnapshot> = vec![];
+        let mut deleted_object_ids: Vec<StoredDeletedObjectSnapshot> = vec![];
+
+        for object in objects {
+            match object {
+                ObjectChangeToCommit::MutatedObject(stored_object) => {
+                    mutated_objects.push(stored_object.into());
+                }
+                ObjectChangeToCommit::DeletedObject(stored_deleted_object) => {
+                    deleted_object_ids.push(stored_deleted_object.into());
+                }
+            }
+        }
+
+        transactional_blocking_with_retry!(
```

Comment:
the schema of objects and objects_snapshot is slightly different, where objects does not include wrapped or deleted rows while objects_snapshot has it and differentiate them via the object status column